src/metabase/pulse/render/body.clj

```diff
@@ -112,12 +120,14 @@
 (s/defn ^:private prep-for-html-rendering
   "Convert the query results (`cols` and `rows`) into a formatted seq of rows (list of strings) that can be rendered as
   HTML"
-  [timezone-id :- (s/maybe s/Str) card {:keys [cols rows]} bar-column max-value column-limit]
-  (let [remapping-lookup (create-remapping-lookup cols)
-        limited-cols (take column-limit cols)]
-    (cons
-     (query-results->header-row remapping-lookup card limited-cols bar-column)
-     (query-results->row-seq timezone-id remapping-lookup limited-cols (take rows-limit rows) bar-column max-value))))
+  ([timezone-id card data column-limit]
+   (prep-for-html-rendering timezone-id card data nil nil nil column-limit))
+  ([timezone-id :- (s/maybe s/Str) card {:keys [cols rows]} bar-column min-value max-value column-limit]
+   (let [remapping-lookup (create-remapping-lookup cols)
+         limited-cols (take column-limit cols)]
+     (cons
+      (query-results->header-row remapping-lookup card limited-cols bar-column)
+      (query-results->row-seq timezone-id remapping-lookup limited-cols (take rows-limit rows) bar-column min-value max-value)))))
```

Comment:
This function is getting kinda hairy with 7 positional args. I'd say once you start getting past 5 you should seriously consider it to taking key-value args

Reply:
@camsaul I changed it (and the related `prep-for-html-rendering`) to use kwargs for `bar-column, min-value, max-value`. Schema doesn't support `& {:keys ...}`, which made it a little jankier than I'd like. I guess it's better. Ultimately I'd like to have a more serious refactor to sidestep needing to shuffle around seven independent bits of data, but that seems out of scope for this PR, especially since pulse stuff is being worked on for dashboard subscriptions